router/http/cards/routes.go

```diff
@@ -0,0 +1,69 @@
+package cards
+
+import (
+  "net/http"
+  "strconv"
+
+  "github.com/gin-gonic/gin"
+
+  "github.com/gbrayhan/academy-go-q12021/domain/card"
+  domainErrors "github.com/gbrayhan/academy-go-q12021/domain/errors"
+)
+
+// NewRoutesFactory create and returns a factory to create routes for the card
+func NewRoutesFactory(group *gin.RouterGroup) func(service card.CardService) {
+  cardRoutesFactory := func(service card.CardService) {
+    group.GET("/", func(c *gin.Context) {
```

Comment:
Maybe we can create separate functions for each of the groups. That way the definition of the factory wouldn't get too big. What do you think?

Reply:
I think that if we create files for those controllers it only increases the complexity of the folders, the philosophy is that the controllers have as few lines as possible

Reply:
Oh, not files. I meant something like this:

```
func foo(c *gin.Context) {...}

group.GET("/", foo)
```

I guess you'd need to send more parameters than only the context. But it's up to you since the current way also looks like a common pattern.